### PR TITLE
rgw: modify realm entities init resolution

### PR DESCRIPTION
**Pull request by yehudasa**
- Try to identify on initialization (`radosgw`, `radosgw-admin`)  the correct realm and zonegroup if only zone is provided. Do it even if default realm and zonegroups are configured and make sure that provided params are consistent -- don't allow zone to run under the wrong zonegroup.
 - Add new configurables that to be able to start `radosgw` using explicit entity IDs (`rgw_zone_id`, `rgw_zonegroup_id`, `rgw_realm_id`). These apply to `radosgw-admin` too, but old params that were previously used there are retained for backward compatibility.
 - Simplify `svc.zone` initialization
 - Require explicit params for certain realm/zonegroup/zone operations (don't rely on defaults)

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by cbodley — APPROVED** [review]
looks sensible, but i do worry about regressions in our documented multisite setup process. is it passing multisite tests in teuthology?

**Comment by yehudasa** [comment]
@cbodley it passed everything a few weeks ago, I'll run it again after rebase

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -2999,6 +2999,150 @@ class JSONFormatter_PrettyZone : public JSONFormatter {
...
+
+static int try_to_resolve_local_entities(string& realm_id, string& realm_name,
+                                         string& zonegroup_id, string& zonegroup_name,
+                                         string& zone_id, string& zone_name)
+{
+  /*
+   * Try to figure out realm, zonegroup, and zone entities, based on provided params and local zone.
+   *
+   * First read the local zone info (for zone id/name). Then search existing realm and period
+   *  configuration and if found, update (but don't override) passed params.
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc:3085]: sorry to dig up old work, but i've been doing some refactoring and had a question:

why is it necessary to search all realms here? if `try_to_resolve_local_zone()` successfully reads `RGWZoneParams`, can't we use its `realm_id` to load its realm directly?

Comment (cbodley) [reply src/rgw/rgw_admin.cc:3085]: proposing this change in https://github.com/ceph/ceph/pull/47762

**Pull request merged**